Language/Statements/Switch/type_t01.dart

```diff
@@ -2,23 +2,18 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion It is a static warning if the type of e is may not be assigned to
+/// @assertion It is not an error if the type of e is may not be assigned to
 /// the type of ek.
-/// @description Checks that a compile error is produced if the type of the
-/// switch expression's value cannot be assigned to any of the case expression
-/// types.
+///
+/// @description Checks that no error is produced if the type of the switch
+/// expression's value cannot be assigned to any of the case expression types.
 /// @author vasya
 
+// SharedOptions=--enable-experiment=patterns
 
 main() {
-  switch (false) {
+  switch ("String") {
     case 0:
-//       ^
-// [analyzer] unspecified
-// [cfe] unspecified
     case 1:
```

Comment:
Do we actually have any rules about how to treat a switch statement where the body is just one sequence of cases with no body? It's not even a single case group, because not even the last case has anything in its body.

It might be considered to be similar to a switch which is non-exhaustive (in situations where we know that kind of thing), but if we keep this test then we have decided that it's OK.

OK, having thought about this for a bit: Seems like we have been doing that for years, so let's just keep it like that!